prql-compiler/src/sql/std.sql.prql

```diff
@@ -38,8 +38,9 @@ let read_parquet = source -> s"read_parquet({source})"
 let read_csv = source -> s"read_csv_auto({source})"
 
 let mul = l r -> null
-let div = l r -> null
-let mod = l r -> null
+let div_i = l r -> s"FLOOR(FLOOR(ABS({l} / {r})) * SIGN({l} / {r}))"
+let div_f = l r -> s"({l} * 1.0 / {r})"
```

Comment:
@aljazerzen how would you feel about:
- Generic dialect is just `l / r`
- MySQL & MsSql use `s"({l} * 1.0 / {r})"`

That way, the default SQL is cleaner, and we keep the guarantees when a dialect is specified.

(I'm happy to make the change if you agree)

Reply:
I don't like that. Character `/` in PRQL means something else than `/` in SQL. I don't think it should be problematic to emit code that looks differently.

All of this would be way more ergonomic if we displayed query results in the book, instead of SQL.

Reply:
> I don't like that. Character `/` in PRQL means something else than `/` in SQL.

I agree with the principle — if it's different then that's bad.
But it's only different in some minor dialects. My suggestion is to support those iff they're explicitly specified.

Another approach would be to default to DuckDB dialect (or Postgres, or Snowflake, i.e. one which has reasonable behavior here), rather than have the default try to cater to every dialect. Then we could use `/` safely.

> All of this would be way more ergonomic if we displayed query results in the book, instead of SQL.

Showing results would be useful too, but many people are much more used to SQL, and it's much clearer to them to see the actual SQL rather than see the results of a query on a table they're not familiar with.

I get that there's a tradeoff there — to what extent are we producing SQL vs producing query results. I have heard this as a benefit of PRQL over comparable langs. I think possibly we're on different sides of this discussion :)

Reply:
If we could be emitting simpler SQL that still passes the tests, we should definitely simply it. This is not hard now.

I don't strongly feel we should switch to showing query results instead of SQL, just a suggestion.

But I do feel strongly that if we decide on different semantics than SQL, we should compile to something that does implement those semantics correctly, in all edge cases. Otherwise our argument that "PRQL can be used with any SQL database" is simply not true.

We could present this as "PRQL handles edge cases for you".

Reply:
I think we agree on the principles:
- PRQL should be correct (not approximately correct)
- We should have legible SQL

...and here they're in conflict. 

I probably weigh Legibility more highly than you?

Is there a synthesis of these tradeoffs where we can do both? 
- Default compile to DuckDB dialect?
- Default compile to DuckDB on the website & playground?
- Have a small list of supported dialects that work with the generic dialect, and others need to explicitly specify the dialect

WDYT?